stdlib/asm/math/u64.masm

```diff
@@ -604,3 +604,89 @@ export.rotr
     not
     cswap
 end
+
+#! Counts the number of leading zeros of one unsigned 64-bit integer.
+#! The input value is assumed to be represented using 32 bit limbs, but this is not checked.
+#! Stack transition looks as follows:
+#! [n_hi, n_lo, ...] -> [clz, ...], where clz is a number of leading zeros of value n.
+#! This takes 59 cycles if n < 2^32 and 32 cycles otherwise.
+export.clz
+    u32clz
+    dup.0
+    push.32
+    eq
+
+    if.true
+        swap
+        u32clz
+        add
+    else
+        swap
+        drop
+    end
```

Comment:
Usage of the `if` construction here can save 27 cycles (or 32 cycles in case of `clo`/`cto`) about half of the time. But `if` block itself adds some amount of cycles each time, although I'm not sure exactly how many, so maybe it is better to change it to the variant which uses `cswap` instruction.

Reply:
I think using an `if` statement here is fine (and it adds 2 cycles, though there are other costs as well) - but why not change this to work like so:
```
if n_hi == 0
    return clz(n_lo) + 32
else
    return clz(n_hi)
end
```
This would require a single 32-bit `clz` in all case - or am I missing something?